confluent_kafka/src/Producer.c

```diff
@@ -251,7 +293,8 @@ static PyObject *Producer_produce (Handle *self, PyObject *args,
 	int value_len = 0, key_len = 0;
 	int partition = RD_KAFKA_PARTITION_UA;
 	PyObject *dr_cb = NULL, *dr_cb2 = NULL, *partitioner_cb = NULL;
-	rd_kafka_topic_t *rkt;
+        long long timestamp = 0;
```

Comment:
librdkafka uses 0 as a sentinel in which case it will use the current wall clock time.

I believe LogAppend vs LogCreate time should be enforced on the broker only (as is currently the case), and let supporting clients always send their LogCreateTime.
It seems to me to be the most forgiving scenario.